src/lib/ai.ts

```diff
@@ -297,49 +306,45 @@ ${func.name}(${JSON.stringify(data, null, 2)})\n\`\`\`\n`;
     signal: controller.signal,
   };
 
-  let responsePromise;
-  if (streaming) {
-    responsePromise = openai.chat.completions
-      .create(
-        chatCompletionParams as OpenAI.Chat.ChatCompletionCreateParamsStreaming,
-        chatCompletionReqOptions
-      )
-      .then(async (streamResponse) => {
-        for await (const streamChunk of streamResponse) {
-          const parsedData = parseOpenAIChunkResponse(streamChunk);
-          await streamOpenAIResponse(
-            parsedData.token,
-            parsedData.functionName,
-            parsedData.functionArgs
-          );
-        }
-
-        const content = buffer.join("");
-        return handleOpenAIResponse(content, functionName, functionArgs);
-      })
-      .catch((err) => {
-        return handleError(err);
-      })
-      .finally(() => {
-        removeEventListener("keydown", handleCancel);
-      });
-  } else {
-    responsePromise = openai.chat.completions
-      .create(
-        chatCompletionParams as OpenAI.Chat.CompletionCreateParamsNonStreaming,
-        chatCompletionReqOptions
-      )
-      .then(async (response: OpenAI.Chat.ChatCompletion) => {
-        const { content, functionName, functionArgs } = parseOpenAIResponse(response);
-        return handleOpenAIResponse(content, functionName, functionArgs);
-      })
-      .catch((err) => {
-        return handleError(err);
-      })
-      .finally(() => {
-        removeEventListener("keydown", handleCancel);
-      });
-  }
+  const handleStreamingResponse: ResponseHandlerStrategy = async (
+    streamResponse: Stream<ChatCompletionChunk>
+  ) => {
+    for await (const streamChunk of streamResponse) {
+      const parsedData = parseOpenAIChunkResponse(streamChunk);
+      await streamOpenAIResponse(
+        parsedData.token,
+        parsedData.functionName,
+        parsedData.functionArgs
+      );
+    }
+
+    const content = buffer.join("");
+    return handleOpenAIResponse(content, functionName, functionArgs);
+  };
+
+  const handleNonStreamingResponse: ResponseHandlerStrategy = async (response: ChatCompletion) => {
+    const { content, functionName, functionArgs } = parseOpenAIResponse(response);
+    return handleOpenAIResponse(content, functionName, functionArgs);
+  };
+
+  const responseHandler: ResponseHandlerStrategy = streaming
```

Comment:
Rename this to a verb (it's active, it will "handle the response" vs. passive "responseHandler").  Also, infer the type:

```suggestion
  const handleResponse = streaming
```

Reply:
Done